### ISSUE TITLE
Problem with using new URL in Image.tsx

### ISSUE THREAD

**Issue by moritzjacobs** [datocms/react-datocms#69]
Hi! Thanks for your work on this package!

I was updating to v4 and wanted to try to reduce the gql data from our query and let the component handle the srcSet building, alas it produces a problem for us:

You're using a whatwg URL to [modify search params](https://github.com/datocms/react-datocms/blob/master/src/Image/index.tsx#L160-L178) on a src URL. In our projects we're masking the Dato CDN behind a Next.js rewrite, so if the CDA returns `https://www.datocms-assets.com/12345/1234567890-foobar.jpg` we rewrite that to  `/cdn/12345/1234567890-foobar.jpg`. Next.js handles that like a charm but in v4 now the `new URL(src)` doesn't work, because URL only knows how to handle absolute URLs.

I was trying to make this a PR but couldn't get it to build correctly, but something like this should to the trick (untested!):

```ts
const buildSrcSet = (

  // … snip …

  const pseudoUrl = `https://example.com`;
  
  return candidateMultipliers
    .map((multiplier) => {
      let url: URL | undefined = undefined;

      try {
        url = new URL(src);
      } catch (error) {
        // seemingly invalid URL => maybe relative path?
        if (error instanceof TypeError) {
          // join with trimmed slashes
          url = new URL([pseudoUrl, src.replace(/^\/|\/$/g, "")].join("/"));
        } else {
          throw error;
        }
      }

      // … snip …

      return `${url.toString().replace(pseudoUrl, "")} ${finalWidth}w`;
    })
    .filter(Boolean)
    .join(',');
};
```

Thanks for your consideration!

**Comment by stefanoverna** [comment]
Fixed in 4.0.7! :+1:

**Comment by moritzjacobs** [comment]
Thank you!
